### PR TITLE
Header for PaginatedDataTable

### PR DESCRIPTION
**Pull request by Hixie**
Also renames ButtonTheme.footer to ButtonTheme.bar.

### CODE REVIEW

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 1]
```diff
@@ -237,7 +266,41 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
 
   @override
   Widget build(BuildContext context) {
-    final TextStyle textStyle = Theme.of(context).textTheme.caption;
+    // TODO(ianh): This whole build function doesn't handle RTL yet.
+    ThemeData theme = Theme.of(context);
```
Comment (abarth) [inline packages/flutter/lib/src/material/paginated_data_table.dart]: We're inconsistent about how we name this variable.  Current thought is to use `themeData`.

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 2]
```diff
@@ -237,7 +266,41 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
...
+        // TODO(ianh): Better magic. See https://github.com/flutter/flutter/issues/4460
+        leftPadding = 12.0;
+      }
+    } else if (_selectedRowCount == 1) {
+      // TODO(ianh): Real l10n.
+      headerWidgets.add(new Flexible(child: new Text('1 item selected')));
+    } else {
+      headerWidgets.add(new Flexible(child: new Text('$_selectedRowCount items selected')));
+    }
+    if (config.actions != null)
```
Comment (abarth) [inline packages/flutter/lib/src/material/paginated_data_table.dart]: Use `{` for if statements with more than one line in their body.

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/material/paginated_data_table.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/paginated_data_table.dart]: fixed

File: packages/flutter/lib/src/material/paginated_data_table.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/paginated_data_table.dart]: fixed

**Pull request merged**